pkg/apis/acid.zalan.do/v1/crds.go

```diff
@@ -247,6 +247,91 @@ var PostgresCRDResourceValidation = apiextv1.CustomResourceValidation{
 							"user": {
 								Type: "string",
 							},
+							"nodeAffinity": {
```

Comment:
gosh, that's a lot of lines. Is this validation covering all affinity options? Maybe we just cut it short with:
```
"nodeAffinity": {
	Type:                   "object",
	XPreserveUnknownFields: util.True(),
},
```

Reply:
at least, makes it easier to copy it to the CRD yaml [manifest](https://github.com/zalando/postgres-operator/blob/master/manifests/postgresql.crd.yaml) where this part is still missing, I think

Reply:
@FxKu for some reason, it was just kept trimming extra fields, even with `XPreserveUnknownFields`. That is why I had to expand it almost fully. :(

Reply:
I just tested it again, and yes, even with `XPreserveUnknownFields` it still trims `unknown` fields and completely breaks node affinity support. Btw, before Nov 17th, it used to work fine.

Reply:
ok, can you remove this part from the validation as it covers the anti_affinity of the pooler

Reply:
sounds good! will remove